common/telemetry/grpc.go

```diff
@@ -134,6 +135,17 @@ func (c *customServerStatsHandler) HandleRPC(ctx context.Context, stat stats.RPC
 	c.wrapped.HandleRPC(ctx, stat)
 
 	switch s := stat.(type) {
+	case *stats.InHeader:
+		if c.isDebug {
+			span := trace.SpanFromContext(ctx)
+			for key, values := range s.Header {
+				span.SetAttributes(attribute.StringSlice("rpc.request.headers."+key, values))
+			}
+			deadline, ok := ctx.Deadline()
+			if ok {
+				span.SetAttributes(attribute.String("rpc.request.headers.timeout", deadline.Format(time.RFC3339Nano)))
+			}
```

Comment:
The timeout is not part of the grpc metadata, but part of the HTTP2 headers. This seems to be the best way to access it.

Reply:
Is it possible that `"timeout"` is already one of the keys?

Reply:
I haven't see it, despite having a timeout set. I can't find a canonical list of grpc metadata keys, though.

Reply:
I could add a prefix?